### PR TITLE
Yihgu/refactor api

### PR DESCRIPTION
**Pull request by YihuiGuo**
Add framework for Feathr API layer.

### CODE REVIEW

File: feathr_project/feathr/api/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+azure-core
```
Comment (hangfei) [inline feathr_project/feathr/api/requirements.txt:1]: who/what will use this file?

**Review by hangfei — COMMENTED** [review]
A more detailed description of this PR is needed for future references.

**Comment by YihuiGuo** [comment]
Add link to design document :
https://microsoft.sharepoint.com/:w:/t/AICustomerEngineeringTeam/ESCFZIcBrpxBmkITXDwT-7EB9dB_V9F-eStOqcFPh1ZfOg

**Comment by hangfei** [comment]
Please merge with latest.

This PR is pretty big. Let's follow https://github.com/linkedin/feathr/blob/main/docs/dev_guide/pull_request_guideline.md so we can refer to the documents in the future.

**Comment by hangfei** [comment]
> Add link to design document : https://microsoft.sharepoint.com/:w:/t/AICustomerEngineeringTeam/ESCFZIcBrpxBmkITXDwT-7EB9dB_V9F-eStOqcFPh1ZfOg

Could you put this into your description section.

**Comment by hangfei** [comment]
Since this PR has some conflicts with main, let me know when you merged so I can start review.

Please ping me once you are ready.

**Pull request closed without merging**